src/main/java/com/zaxxer/hikari/metrics/prometheus/PrometheusMetricsTracker.java

```diff
@@ -17,41 +17,52 @@
 package com.zaxxer.hikari.metrics.prometheus;
 
 import com.zaxxer.hikari.metrics.IMetricsTracker;
-import io.prometheus.client.Collector;
 import io.prometheus.client.CollectorRegistry;
 import io.prometheus.client.Counter;
 import io.prometheus.client.Summary;
-import java.util.concurrent.TimeUnit;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 
 class PrometheusMetricsTracker implements IMetricsTracker
 {
    private static final Counter CONNECTION_TIMEOUT_COUNTER = Counter.build()
       .name("hikaricp_connection_timeout_total")
       .labelNames("pool")
       .help("Connection timeout total count")
-      .register();
+      .create();
    private static final Summary ELAPSED_ACQUIRED_SUMMARY = Summary.build()
       .name("hikaricp_connection_acquired_nanos")
       .labelNames("pool")
       .help("Connection acquired time (ns)")
-      .register();
+      .create();
    private static final Summary ELAPSED_BORROWED_SUMMARY = Summary.build()
       .name("hikaricp_connection_usage_millis")
       .labelNames("pool")
       .help("Connection usage (ms)")
-      .register();
+      .create();
    private static final Summary ELAPSED_CREATION_SUMMARY = Summary.build()
       .name("hikaricp_connection_creation_millis")
       .labelNames("pool")
       .help("Connection creation (ms)")
-      .register();
+      .create();
+
+   private static final Set<CollectorRegistry> registered = Collections.newSetFromMap(new ConcurrentHashMap<>());
 
    private final Counter.Child connectionTimeoutCounterChild;
    private final Summary.Child elapsedAcquiredSummaryChild;
    private final Summary.Child elapsedBorrowedSummaryChild;
    private final Summary.Child elapsedCreationSummaryChild;
 
-   PrometheusMetricsTracker(String poolName) {
+   PrometheusMetricsTracker(String poolName, CollectorRegistry registry) {
+      if (!registered.contains(registry)) {
+         CONNECTION_TIMEOUT_COUNTER.register(registry);
+         ELAPSED_ACQUIRED_SUMMARY.register(registry);
+         ELAPSED_BORROWED_SUMMARY.register(registry);
+         ELAPSED_CREATION_SUMMARY.register(registry);
+         registered.add(registry);
```

Comment:
Not thread-safe, despite use of a wrapped ``ConcurrentHashMap``.  ``registered.contains()`` and ``registered.add()`` are not collectively atomic.

Reply:
These functions are rarely called.  Just use a simple set and synchronize the method.